tests/fixtures/simple/app/api/slow-not-cacheable-with-html-read/route.js

```diff
@@ -0,0 +1,22 @@
+import { NextResponse } from 'next/server'
+import { readFile } from 'node:fs/promises'
+
+export async function GET() {
+  // This adds intentional delay here to make it more likely to hit some next-server
+  // initialization side-effects such as preloading page entries
+  // and trying to assert that side-effects do NOT impact the response.
+  // There is no way to force problematic side-effect scenario to happen without
+  // modifying the next internals.
+  // See https://github.com/vercel/next.js/blob/592401bb7fec83079716b2c9b090db580a63483f/packages/next/src/server/next-server.ts#L321-L327
+  // which starts NOT awaited async work
+  await new Promise((resolve) => setTimeout(resolve, 5_000))
```

Comment:
without some kind of sleep in function, it was VERY hit'n'miss wether issue was reproducable - with this sleep, at this locally it was always reprodicable - but because the whole issue is timing/async - this is not 100% test scenario, but I couldn't figure out anything better without doing some kind of modifications to next.js code itself